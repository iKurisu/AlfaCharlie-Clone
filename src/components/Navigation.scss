@import 'sass/_variables.scss';

.nav {
  width: 30px;
  height: 80vh;
  position: fixed;
  right: 46px;
  top: 10vh;
  z-index: 10;
  display: flex;
  align-items: center;
  justify-content: center;
}

.nav ul {
  transform: rotate(90deg);
  list-style: none;
  white-space: nowrap;
  line-height: 15px;
}

.nav li {
  display: inline-block;
  cursor: pointer;
}

.nav .link-wrapper {
  margin: 0 2vh;
  display: inline-block;
  position: relative;
}

.nav .link-name {
  display: block;
  margin: 0 1vh;
  color: $grey;
  text-transform: uppercase;
  font-size: 10px;
  letter-spacing: 6px;
  transition: color 0.3s;
  line-height: 10px;
}

.nav li:hover .link-name {
  color: $red;
}

.nav .active .link-name {
  color: $red;
}

@media (max-width: $l) {
  .nav {
    display: none;
  }
}

@media (min-width: $xxl) {
  .nav {
    right: 57px;
  }

  .nav .link-name {
    font-size: 12px;
    letter-spacing: 7.05px;
  }
}
