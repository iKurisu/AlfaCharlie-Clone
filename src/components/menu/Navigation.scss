.menu-nav-wrapper {
  position: relative;
  z-index: 10;
  height: 100%;
  width: 50%;
}

.-flex {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
