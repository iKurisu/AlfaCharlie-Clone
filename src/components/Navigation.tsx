import React from "react";
import "./Navigation.scss";

interface NavProps {
  links: [string, boolean][];
}

const Navigation = ({ links }: NavProps): JSX.Element => (
  <nav className="nav">
    <ul>
      {links.map(
        ([link, active], id): JSX.Element => (
          <li key={id}>
            <span className={`link-wrapper${active ? " active" : ""}`}>
              <span className="link-name">{link}</span>
            </span>
          </li>
        )
      )}
    </ul>
  </nav>
);

export default Navigation;
