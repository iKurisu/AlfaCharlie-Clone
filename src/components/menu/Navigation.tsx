import React from "react";
import Link from "./navigation/Link";
import "./Navigation.scss";

const Navigation = (): JSX.Element => {
  const links = ["agency", "work", "journal", "contact"];

  return (
    <div className="menu-nav-wrapper -flex">
      <nav className="menu-nav-links -flex">
        {links.map(
          (link: string, id: number): JSX.Element => (
            <Link link={link} key={id} />
          )
        )}
      </nav>
    </div>
  );
};

export default Navigation;
