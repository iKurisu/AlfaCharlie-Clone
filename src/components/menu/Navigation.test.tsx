import React from "react";
import renderer from "react-test-renderer";
import Navigation from "./Navigation";

it("renders correctly", (): void => {
  const component = renderer.create(<Navigation />);

  expect(component).toMatchSnapshot();
});
