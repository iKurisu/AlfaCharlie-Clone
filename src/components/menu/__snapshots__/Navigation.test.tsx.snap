// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="menu-nav-wrapper -flex"
>
  <nav
    className="menu-nav-links -flex"
  >
    <a
      className="menu-nav-link"
    >
      <span
        className="link-characters"
      >
        <span
          className="link-character"
        >
          a
        </span>
        <span
          className="link-character"
        >
          g
        </span>
        <span
          className="link-character"
        >
          e
        </span>
        <span
          className="link-character"
        >
          n
        </span>
        <span
          className="link-character"
        >
          c
        </span>
        <span
          className="link-character"
        >
          y
        </span>
      </span>
    </a>
    <a
      className="menu-nav-link"
    >
      <span
        className="link-characters"
      >
        <span
          className="link-character"
        >
          w
        </span>
        <span
          className="link-character"
        >
          o
        </span>
        <span
          className="link-character"
        >
          r
        </span>
        <span
          className="link-character"
        >
          k
        </span>
      </span>
    </a>
    <a
      className="menu-nav-link"
    >
      <span
        className="link-characters"
      >
        <span
          className="link-character"
        >
          j
        </span>
        <span
          className="link-character"
        >
          o
        </span>
        <span
          className="link-character"
        >
          u
        </span>
        <span
          className="link-character"
        >
          r
        </span>
        <span
          className="link-character"
        >
          n
        </span>
        <span
          className="link-character"
        >
          a
        </span>
        <span
          className="link-character"
        >
          l
        </span>
      </span>
    </a>
    <a
      className="menu-nav-link"
    >
      <span
        className="link-characters"
      >
        <span
          className="link-character"
        >
          c
        </span>
        <span
          className="link-character"
        >
          o
        </span>
        <span
          className="link-character"
        >
          n
        </span>
        <span
          className="link-character"
        >
          t
        </span>
        <span
          className="link-character"
        >
          a
        </span>
        <span
          className="link-character"
        >
          c
        </span>
        <span
          className="link-character"
        >
          t
        </span>
      </span>
    </a>
  </nav>
</div>
`;
